src/System.Linq/src/System/Linq/Lookup.cs

```diff
@@ -182,7 +196,7 @@ internal List<TResult> ToList<TResult>(Func<TKey, IEnumerable<TElement>, TResult
             return _count;
         }
 
-        public IEnumerable<TResult> ApplyResultSelector<TResult>(Func<TKey, IEnumerable<TElement>, TResult> resultSelector)
+        public IEnumerator<TResult> ApplyResultSelector<TResult>(Func<TKey, IEnumerable<TElement>, TResult> resultSelector)
```

Comment:
After thinking about it and considering that eumerable here is really just a bit of extra data over enumerator (method has just one parameter), it might not worth the trouble and just use IEnumerable. The gain is indeed pretty small.